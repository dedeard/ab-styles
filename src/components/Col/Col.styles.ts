import { css } from '@emotion/react';

import { config } from '../../config';
import { responsive } from '../../utils';

import type { ColProps } from '.';
import type { DefaultTheme, StyleProps } from '../../types/emotion';

export const base = ({ theme }: StyleProps) => css`
  position: relative;
  flex: 1 0 0%;

  width: 100%;
  max-width: 100%;

  box-sizing: border-box;

  padding-left: ${config(theme).grid.space / 2}rem;
  padding-right: ${config(theme).grid.space / 2}rem;
`;

export const size = (props: ColProps & StyleProps) =>
  responsive(
    config(props.theme).grid.breakpoints,
    (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
      props[breakpoint] &&
      `
        flex: 0 0 ${
          (props[breakpoint] / config(props.theme).grid.columns[breakpoint]) *
          100
        }%;
        max-width: ${
          (props[breakpoint] / config(props.theme).grid.columns[breakpoint]) *
          100
        }%;
    `
  );

export const offset = ({ theme, offset }: ColProps & StyleProps) =>
  offset &&
  responsive(
    config(theme).grid.breakpoints,
    (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
      offset[breakpoint] &&
      `
        margin-left: ${
          offset[breakpoint] >= 0
            ? (offset[breakpoint] / config(theme).grid.columns[breakpoint]) *
              100
            : 0
        }%;
    `
  );
