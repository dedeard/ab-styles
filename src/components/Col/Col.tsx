import { CSSProperties, ReactNode } from 'react';

import styled from '@emotion/styled';
import { css } from '@emotion/react';

import { config, BREAKPOINTS } from '../../config';
import { media } from '../../utils';

import { Breakpoints, StyleProps } from '../../types/emotion';

export interface ColProps {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  offset?: Record<Breakpoints, number>;
  style?: CSSProperties;
  className?: string;
  children: ReactNode;
}

const baseStyle = ({ theme }: StyleProps) => css`
  label: col;

  position: relative;
  flex: 1 0 0%;

  width: 100%;
  max-width: 100%;

  box-sizing: border-box;

  ${BREAKPOINTS.map(
    (breakpoint) =>
      css`
        ${media(breakpoint)} {
          padding-left: ${config(theme).grid.gutter[breakpoint] / 2}rem;
          padding-right: ${config(theme).grid.gutter[breakpoint] / 2}rem;
        }
      `
  )}
`;

const sizeStyle = (props: ColProps & StyleProps) => {
  const { theme } = props;

  return css`
    ${BREAKPOINTS.map(
      (breakpoint) =>
        props[breakpoint] &&
        css`
          ${media(breakpoint)} {
            flex: 0 0
              ${(props[breakpoint] / config(theme).grid.columns[breakpoint]) *
              100}%;
            max-width: ${(props[breakpoint] /
              config(theme).grid.columns[breakpoint]) *
            100}%;
          }
        `
    )}
  `;
};

const offsetStyle = ({ theme, offset }: ColProps & StyleProps) =>
  offset &&
  typeof offset === 'object' &&
  BREAKPOINTS.filter((breakpoint) => offset[breakpoint]).map(
    (breakpoint) =>
      offset[breakpoint] &&
      css`
        ${media(breakpoint)} {
          margin-left: ${offset[breakpoint] >= 0
            ? (offset[breakpoint] / config(theme).grid.columns[breakpoint]) *
              100
            : 0}%;
        }
      `
  );

const Col = styled('div')<ColProps>(baseStyle, sizeStyle, offsetStyle);

export default Col;
