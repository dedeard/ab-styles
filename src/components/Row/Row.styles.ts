import { css } from '@emotion/react';

import { config } from '../../config';
import { responsive } from '../../utils';

import type { RowProps } from '.';
import { DefaultTheme, StyleProps } from '../../types/emotion';

export const base = ({ theme }: StyleProps) => css`
  display: flex;
  flex-wrap: wrap;

  box-sizing: border-box;

  margin-left: -${config(theme).grid.space / 2}rem;
  margin-right: -${config(theme).grid.space / 2}rem;
`;

export const align = ({ theme, align }: RowProps & StyleProps) =>
  typeof align === 'object'
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
          align[breakpoint] &&
          `
            align-items: ${align[breakpoint]};
          `
      )
    : css`
        align-items: ${align};
      `;

export const justify = ({ theme, justify }: RowProps & StyleProps) =>
  typeof justify === 'object'
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
          justify[breakpoint] &&
          `
            justify-content: ${justify[breakpoint]};
          `
      )
    : css`
        justify-content: ${justify};
      `;

export const reverse = ({ theme, reverse }: RowProps & StyleProps) =>
  reverse &&
  (Array.isArray(reverse)
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) => `
          flex-direction: ${
            reverse.includes(breakpoint) ? 'row-reverse' : 'row'
          };
        `
      )
    : css`
        flex-direction: row-reverse;
      `);

export const noGutter = ({ noGutters }: RowProps) =>
  noGutters &&
  css`
    .grid-col {
      padding-left: 0;
      padding-right: 0;
    }
  `;
