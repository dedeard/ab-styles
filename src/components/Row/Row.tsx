import { ReactNode } from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/react';

import { config } from '../../config';
import { responsive } from '../../utils';

import { Col } from '../Col';

import {
  Align,
  AlignProps,
  Justify,
  JustifyProps,
  DefaultTheme,
  StyleProps,
} from '../../types/emotion';

export interface RowProps {
  /**
   * The CSS `align-items` property
   */
  align?: Align | AlignProps;

  /**
   * The CSS `justify-content` property
   */
  justify?: Justify | JustifyProps;

  /**
   * The CSS `flex-direction` property
   */
  reverse?: Array<keyof DefaultTheme['grid']['breakpoints']> | boolean;

  /**
   * Remove the gutter(padding) from `Cols`
   */
  noGutters?: boolean;

  /**
   * The children nodes
   */
  children: ReactNode;
}

const baseStyle = ({ theme }: StyleProps) => css`
  label: row;

  display: flex;
  flex-wrap: wrap;

  box-sizing: border-box;

  ${responsive(
    config(theme).grid.breakpoints,
    (breakpoint: keyof DefaultTheme['grid']['breakpoints']) => `
      margin-left: -${config(theme).grid.gutter[breakpoint] / 2}rem;
      margin-right: -${config(theme).grid.gutter[breakpoint] / 2}rem;
    `
  )};
`;

const alignStyle = ({ theme, align }: RowProps & StyleProps) =>
  typeof align === 'object'
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
          align[breakpoint] &&
          `
            align-items: ${align[breakpoint]};
          `
      )
    : css`
        align-items: ${align};
      `;

const justifyStyle = ({ theme, justify }: RowProps & StyleProps) =>
  typeof justify === 'object'
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) =>
          justify[breakpoint] &&
          `
            justify-content: ${justify[breakpoint]};
          `
      )
    : css`
        justify-items: ${justify};
      `;

const reverseStyle = ({ theme, reverse }: RowProps & StyleProps) =>
  reverse &&
  (Array.isArray(reverse)
    ? responsive(
        config(theme).grid.breakpoints,
        (breakpoint: keyof DefaultTheme['grid']['breakpoints']) => `
          flex-direction: ${
            reverse.includes(breakpoint) ? 'row-reverse' : 'row'
          };
        `
      )
    : css`
        flex-direction: row-reverse;
      `);

const noGutterStyle = ({ noGutters }: RowProps) =>
  noGutters &&
  css`
    ${Col} {
      padding-left: 0;
      padding-right: 0;
    }
  `;

const Row = styled('div')<RowProps>(
  baseStyle,
  alignStyle,
  justifyStyle,
  reverseStyle,
  noGutterStyle
);

export default Row;
