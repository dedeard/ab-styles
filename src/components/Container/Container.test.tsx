import { render } from '@testing-library/react';

import { Container } from '.';
import { Row } from '../Row';
import { Col } from '../Col';

describe('Container', () => {
  it('should render with default styles for media %s', () => {
    const { container } = render(<Container>Container</Container>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with fluid styles when passed the fluid prop', () => {
    const { container } = render(<Container fluid>Container fluid</Container>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with debug styles when passed the debug prop', () => {
    const { container } = render(
      <Container debug>
        <Row>
          <Col>Col</Col>
        </Row>
      </Container>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
