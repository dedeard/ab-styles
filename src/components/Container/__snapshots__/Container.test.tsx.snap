// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with debug styles when passed the debug prop 1`] = `
.grid-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-0 {
    max-width: 45rem;
  }
}

@media (min-width: 64em) {
  .grid-0 {
    max-width: 60rem;
  }
}

@media (min-width: 71.25em) {
  .grid-0 {
    max-width: 71.25rem;
  }
}

.grid-0 .grid-3 {
  background: #0470F50D;
}

.grid-0 .grid-5 {
  background: #0470F50D;
  border: 1px solid #0470F5;
}

.grid-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 64em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .grid-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

.grid-4 {
  position: relative;
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .grid-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

.emotion-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-0 {
    max-width: 45rem;
  }
}

@media (min-width: 64em) {
  .emotion-0 {
    max-width: 60rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-0 {
    max-width: 71.25rem;
  }
}

.emotion-0 .emotion-3 {
  background: #0470F50D;
}

.emotion-0 .emotion-5 {
  background: #0470F50D;
  border: 1px solid #0470F5;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .emotion-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .emotion-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 64em) {
  .emotion-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-2 {
    margin-left: -0.9375rem;
    margin-right: -0.9375rem;
  }
}

.emotion-4 {
  position: relative;
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .emotion-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .emotion-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .emotion-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-4 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      Col
    </div>
  </div>
</div>
`;

exports[`Container should render with default styles for media %s 1`] = `
.grid-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-0 {
    max-width: 45rem;
  }
}

@media (min-width: 64em) {
  .grid-0 {
    max-width: 60rem;
  }
}

@media (min-width: 71.25em) {
  .grid-0 {
    max-width: 71.25rem;
  }
}

.emotion-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-0 {
    max-width: 45rem;
  }
}

@media (min-width: 64em) {
  .emotion-0 {
    max-width: 60rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-0 {
    max-width: 71.25rem;
  }
}

<div
  class="emotion-0 emotion-1"
>
  Container
</div>
`;

exports[`Container should render with fluid styles when passed the fluid prop 1`] = `
.grid-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .grid-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

.emotion-0 {
  width: 100%;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media (min-width: 1em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 33.75em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 48em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 64em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

@media (min-width: 71.25em) {
  .emotion-0 {
    padding-left: 0.9375rem;
    padding-right: 0.9375rem;
  }
}

<div
  class="emotion-0 emotion-1"
>
  Container fluid
</div>
`;
